src/pages/Campaign/CampaignButtonWithOptions.tsx

```diff
@@ -0,0 +1,163 @@
+        }
+      })
+    }
+    const data = {
+      wallet: account.toLowerCase(),
+      chainId: selectedCampaign.rewardChainIds,
+      clientCode: 'campaign',
+      ref: refs.join(','),
+    }
+    const response = await axios({
+      method: 'POST',
+      url,
+      data,
+    })
+    if (response.data.code === 200000) {
+      const rewardContractAddress = response.data.data.ContractAddress
+      const encodedData = response.data.data.EncodedData
+      console.log(`rewardContractAddress`, rewardContractAddress)
+      console.log(`encodedData`, encodedData)
+      try {
+        await sendTransaction(rewardContractAddress, encodedData, BigNumber.from(0), transactionResponse => {
+          return onClaimRewardSuccess(transactionResponse, selectedCampaign.name, selectedCampaignLeaderboard)
+        })
+      } catch (err) {
+        console.error(err)
+      }
+    }
+  }
+
+  return (
+    <StyledCampaignButtonWithOptions
+      style={{
+        padding: '12px 58px',
+        minWidth: 'fit-content',
+        height: 'fit-content',
+        lineHeight: '20px',
+        fontWeight: 500,
+        color: theme.textReverse,
+      }}
+      onClick={e => {
+        e.stopPropagation()
+        setIsShowNetworks(prev => !prev)
+      }}
+      disabled={disabled}
+      ref={containerRef}
+    >
+      <Trans>{type === 'enter_now' ? 'Enter now' : 'Claim Rewards'}</Trans>
+      <ChevronDown
+        size="20px"
+        style={{ position: 'absolute', top: '50%', right: '12px', transform: 'translateY(-50%)' }}
+      />
+      {isShowNetworks && (
+        <OptionsContainer style={{ margin: '0 12px', width: 'calc(100% - 24px)' }}>
+          {chainIds.map(chainId => {
+            return (
+              <Flex
+                key={chainId}
+                alignItems="center"
+                onClick={async () => {
+                  if (type === 'enter_now') {
+                    mixpanelHandler(MIXPANEL_TYPE.CAMPAIGN_ENTER_NOW_CLICKED, { campaign_name: campaign?.name })
+                    window.open(campaign?.enterNowUrl + '?networkId=' + chainId)
+                  } else {
+                    await changeNetwork(chainId, claimRewards)
+                  }
+                }}
+              >
+                <img src={NETWORKS_INFO[chainId].icon} alt="Network" style={{ minWidth: '16px', width: '16px' }} />
+                <Text marginLeft="4px" color={theme.subText} fontSize="12px" fontWeight={500} minWidth="fit-content">
+                  <Trans>
+                    {type === 'enter_now' ? 'Swap' : 'Claim'} on {NETWORKS_INFO[chainId].name}
+                  </Trans>
+                </Text>
+              </Flex>
+            )
+          })}
+        </OptionsContainer>
+      )}
+    </StyledCampaignButtonWithOptions>
+  )
+}
+
+const StyledCampaignButtonWithOptions = styled(Button)`
+  position: relative;
+  ${({ theme }) => theme.mediaWidth.upToSmall`
+    ${css`
+      flex: 1;
+    `}
```

Comment:
why so complex bro, didnt only `flex: 1;` solved it?

Reply:
My havit of wrapping it with css.

Reply:
It's good for linting.

Reply:
okay